src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,35 @@
+package main.java.core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RANDOM_COUNTER = 10;
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        String[] figure = {"Circle", "IsoscelesTrapezoid", "Rectangle", "RightTriangle", "Square"};
+        int i = random.nextInt(figure.length);
+        switch (figure[i]) {
+            case "Circle":
+                return new Circle(ColorSupplier.getRandomColor(),
+                        random.nextInt(RANDOM_COUNTER) + 1);
+            case "IsoscelesTrapezoid":
+                return new IsoscelesTrapezoid(ColorSupplier.getRandomColor(),
+                        random.nextInt(RANDOM_COUNTER) + 1,
+                        random.nextInt(RANDOM_COUNTER) + 12,
+                        random.nextInt(RANDOM_COUNTER) + 1);
```

Comment:
let's create private methods for each figure creation and there explicitly create variables to pass to constructor. don't pass an expression to a constructor